assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetailPage.jsx

```diff
@@ -49,6 +49,10 @@ function CheckResultDetailPage() {
 
   const checkDescription = getCheckDescription(catalog, checkID);
 
+  const targetID = isTargetHost(targetType)
+    ? (clusterHosts.find(({ hostname }) => hostname === targetName) || {})?.id
+    : clusterID;
```

Comment:
I think this could be easily refactored using lodash, too bad we still don't have lodash